Create "Embedding in .NET" page (#789)
surrealdb/docs.surrealdb.com: src/content/doc-surrealdb/embedding/dotnet.mdx

It can be extremely useful for testing scenarios, or for small applications that do not require persistence.

```bash
dotnet add package SurrealDb.Embedded.InMemory
```

### Consume the provider as-is

The simplest way to use an in-memory database instance of SurrealDB is to create an instance of the `SurrealDbMemoryClient` class.

```csharp
// highlight-next-line
using var db = new SurrealDbMemoryClient();

const string TABLE = "person";

var person = new Person
{
    Title = "Founder & CEO",
    Name = new() { FirstName = "Tobie", LastName = "Morgan Hitchcock" },
    Marketing = true
};
var created = await db.Create(TABLE, person);
Console.WriteLine(ToJsonString(created));
```

### Consume the provider via Dependency Injection

Following the .NET Dependency Injection pattern, you can register the in-memory provider using the `AddInMemoryProvider` extension method.
This will allow the `SurrealDbClient` to resolve the `mem://` endpoint.

```csharp
var builder = WebApplication.CreateBuilder(args);

var services = builder.Services;
var configuration = builder.Configuration;

// highlight-start
services
  .AddSurreal("Endpoint=mem://")
  .AddInMemoryProvider();
// highlight-end
```

> You can get an in-depth look at Dependency Injection with SurrealDB in .NET [here](/docs/sdk/dotnet/core/dependency-injection).
> 

Reviewer: ```suggestion
Learn more about [Dependency Injection with SurrealDB in .NET](/docs/sdk/dotnet/core/dependency-injection) in the SDK documentation.

```

> ### Next step

Reviewer: ```suggestion

```

```

  </TabItem>
</Tabs>

### Consume the provider via Dependency Injection

<Tabs groupId="file-embedded-modes">
  <TabItem value="rocksdb" label="RocksDB" default>

Following the .NET Dependency Injection pattern, you can register the file provider using the `AddRocksDbProvider` extension method.
This will allow the `SurrealDbClient` to resolve the `rocksdb://` endpoint.

```csharp
var builder = WebApplication.CreateBuilder(args);

var services = builder.Services;
var configuration = builder.Configuration;

// highlight-start
services
  .AddSurreal("Endpoint=rocksdb://data.db")
  .AddRocksDbProvider();
// highlight-end
```

  </TabItem>
  <TabItem value="surrealkv" label="SurrealKV">

Following the .NET Dependency Injection pattern, you can register the file provider using the `AddSurrealKvProvider` extension method.
This will allow the `SurrealDbClient` to resolve the `surrealkv://` endpoint.

```csharp
var builder = WebApplication.CreateBuilder(args);

var services = builder.Services;
var configuration = builder.Configuration;

// highlight-start
services
  .AddSurreal("Endpoint=surrealkv://data.db")
  .AddSurrealKvProvider();
// highlight-end
```

  </TabItem>
</Tabs>

> You can get an in-depth look at Dependency Injection with SurrealDB in .NET [here](/docs/sdk/dotnet/core/dependency-injection).

Reviewer: ```suggestion
Learn more about [Dependency Injection with SurrealDB in .NET](/docs/sdk/dotnet/core/dependency-injection) in the SDK documentation.
```

### Next step

> Once the file provider is configured, you can use the .NET SDK the same way you would with a remote database.
> Please refer to the [.NET client SDK](/docs/sdk/dotnet) documentation to get started with SurrealDB for .NET.

Reviewer: ```suggestion
Once the file provider is configured, you can use the .NET SDK the same way you would with a remote database.
Please refer to the [.NET client SDK](/docs/sdk/dotnet) documentation to get started.
```
